### PR TITLE
Food goal selection algorithm improvement

### PR DESCRIPTION
**Pull request by ksofiyuk**
Hi,
I implemented a new algorithm for food goal selection.
[+] Fixed problem with looping around one point.
[+] Snake moves to cluster of points with largest weight.
[+] If the cluster is large (e.g. dead snake), then turn on the acceleration.

### CODE REVIEW

File: bot.user.js [hunk 1]
```diff
@@ -352,7 +352,7 @@ window.getSnakeWidth = function () {
 window.sortFood = function (a, b) {
     // a.sz & b.sz - size
     // Divide distance by size so bigger food is prioritised over smaller food
-    return a.distance / a.sz - b.distance / b.sz;
+    return a.distance - b.distance;
```
Comment (ChadSki) [inline bot.user.js]: Comments are now inaccurate

**Comment by Seple** [comment]
Interestingly only problem again collision (Enemy avoidance / Collision prevention)

**Comment by ChadSki** [comment]
Unfortunately this is going to conflict with @chancity's [pending change](https://github.com/ErmiyaEskandary/Slither.io-bot/pull/84)

Can we combine the best parts of both approaches?

Edit: testing it out now and it certainly works better than the master branch. Not sure how it compares to chancity's pending change.

**Comment by ermiyaeskandary** [comment]
@ksofiyuk Will test now - according to tests above, this is better than @chancity 's PR _at the current moment_.

**Comment by ermiyaeskandary** [comment]
@ksofiyuk You should have also used the PR template - it's there for a reason

**Comment by chancity** [comment]
I tried his out,  I couldn't tell for sure if it was better.  From what I saw and how it's code they're comparable and use the same technique

**Comment by ermiyaeskandary** [comment]
@chancity But according to @Drflash55 your code lags - lemme test personally now...

**Comment by ChadSki** [comment]
I was going to complain that this version accelerates too often, but then I realized that bestClusterSize is assigned to 40 here. They do seem to be otherwise very similar!

**Comment by ermiyaeskandary** [comment]
Agreed - I can't specifically pick one out of the two

**Comment by ermiyaeskandary** [comment]
It's also cleaned up and more ready to be merged...

**Comment by Seple** [comment]
Please options Speed food - key enable

**Comment by tjorim** [comment]
@Seple You mean an option to turn on or off the acceleration when going after large amounts of points?

**Comment by ermiyaeskandary** [comment]
@Seple Make an issue please.

**Comment by Seple** [comment]
> @Seple Masz na myśli opcję, aby włączyć lub wyłączyć przyspieszenie idąc po dużej ilości punktów? 

Yes this problem at discussion #87 ,and new problem improvement detect (speed) colision

**Pull request merged**